src/Octoshift/Services/AdoApi.cs

```diff
@@ -523,34 +523,59 @@ public virtual async Task ShareServiceConnection(string adoOrg, string adoTeamPr
+        string apiUrl, webUrl, cloneUrl, branchesUrl, refsUrl, manageUrl;
+        if (!string.IsNullOrWhiteSpace(targetApiUrl))
+        {
+            var apiUri = new Uri(targetApiUrl.TrimEnd('/'));
+            var webHost = apiUri.Host.StartsWith("api.") ? apiUri.Host.Substring(4) : apiUri.Host;
+            var webScheme = apiUri.Scheme;
+            var webBase = $"{webScheme}://{webHost}";
+            apiUrl = $"{targetApiUrl.TrimEnd('/')}/repos/{githubOrg.EscapeDataString()}/{githubRepo.EscapeDataString()}";
+            webUrl = $"{webBase}/{githubOrg.EscapeDataString()}/{githubRepo.EscapeDataString()}";
+            cloneUrl = $"{webBase}/{githubOrg.EscapeDataString()}/{githubRepo.EscapeDataString()}.git";
+            branchesUrl = $"{targetApiUrl.TrimEnd('/')}/repos/{githubOrg.EscapeDataString()}/{githubRepo.EscapeDataString()}/branches";
+            refsUrl = $"{targetApiUrl.TrimEnd('/')}/repos/{githubOrg.EscapeDataString()}/{githubRepo.EscapeDataString()}/git/refs";
+            manageUrl = webUrl;
+        }
+        else
+        {
+            apiUrl = $"https://api.github.com/repos/{githubOrg.EscapeDataString()}/{githubRepo.EscapeDataString()}";
+            webUrl = $"https://github.com/{githubOrg.EscapeDataString()}/{githubRepo.EscapeDataString()}";
+            cloneUrl = $"https://github.com/{githubOrg.EscapeDataString()}/{githubRepo.EscapeDataString()}.git";
+            branchesUrl = $"https://api.github.com/repos/{githubOrg.EscapeDataString()}/{githubRepo.EscapeDataString()}/branches";
+            refsUrl = $"https://api.github.com/repos/{githubOrg.EscapeDataString()}/{githubRepo.EscapeDataString()}/git/refs";
+            manageUrl = webUrl;
+        }
+
         var newRepo = new
         {
             properties = new
             {
-                apiUrl = $"https://api.github.com/repos/{githubOrg.EscapeDataString()}/{githubRepo.EscapeDataString()}",
-                branchesUrl = $"https://api.github.com/repos/{githubOrg.EscapeDataString()}/{githubRepo.EscapeDataString()}/branches",
-                cloneUrl = $"https://github.com/{githubOrg.EscapeDataString()}/{githubRepo.EscapeDataString()}.git",
+                apiUrl,
+                branchesUrl,
+                cloneUrl,
                 connectedServiceId,
                 defaultBranch,
                 fullName = $"{githubOrg}/{githubRepo}",
-                manageUrl = $"https://github.com/{githubOrg.EscapeDataString()}/{githubRepo.EscapeDataString()}",
+                manageUrl,
                 orgName = githubOrg,
-                refsUrl = $"https://api.github.com/repos/{githubOrg.EscapeDataString()}/{githubRepo.EscapeDataString()}/git/refs",
+                refsUrl,
                 safeRepository = $"{githubOrg.EscapeDataString()}/{githubRepo.EscapeDataString()}",
                 shortName = githubRepo,
                 reportBuildStatus = true
             },
             id = $"{githubOrg}/{githubRepo}",
             type = "GitHub",
```

Comment:
Even though this worked for me, I am not 100% certain this does not need to be a different value if the repository is on GHES or GHEC with Data Residency

Reply:
The types in the docs you linked is the type of _build definition_. The type being used here is the type of _repository_ (https://learn.microsoft.com/en-us/rest/api/azure/devops/build/definitions/update?view=azure-devops-rest-6.0#buildrepository)